interfaces/inc/c_pal/threadpool.h

```diff
@@ -22,21 +22,20 @@ extern "C" {
 
 typedef struct THREADPOOL_TAG THREADPOOL;
 typedef struct TIMER_INSTANCE_TAG* TIMER_INSTANCE_HANDLE;
-typedef struct THREADPOOL_WORK_ITEM_TAG* THREADPOOL_WORK_ITEM_HANDLE;
+typedef struct THREADPOOL_WORK_ITEM_CONTEXT_TAG THREADPOOL_WORK_ITEM_CONTEXT;
```

Comment:
poetry: I think we should remove the word "CONTEXT" here to be more consistent with how we name things (e.g. `THANDLE(THREADPOOL)` is not `THANDLE(THREADPOOL_CONTEXT)`.

Generally "context" is used in our code for some collection of data to pass along with a callback.

Reply:
Thanks for the context :)